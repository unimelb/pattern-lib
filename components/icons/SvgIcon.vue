<template>
  <svg
    :width="width"
    :height="height"
    :aria-label="name"
    focusable="false"
    role="img">
    <use :xlink:href="'#icon-' + name" />
    <title>{{ name | capitalize }}</title>
    <slot />
  </svg>
</template>

<script>
// TODO: investigate refactoring to import svgs dynamically using <path> or base64 in CSS
export default {
  name: 'SvgIcon',
  filters: {
    capitalize(value) {
      const startFromCharIndex = 0;

      let capitalisation = value;
      if (!capitalisation) {
        return '';
      }
      capitalisation = capitalisation.toString();

      return capitalisation.charAt(startFromCharIndex).toUpperCase() + capitalisation.slice(1);
    },
  },
  props: {
    name: {
      type: String,
      default: '',
    },
    width: {
      type: [String, Number],
      default: '0',
    },
    height: {
      type: [String, Number],
      default: '0',
    },
  },
};
</script>
