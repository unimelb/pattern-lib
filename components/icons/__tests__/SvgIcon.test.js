import {
  shallow,
} from 'vue-test-utils';
import { axe, toHaveNoViolations } from 'jest-axe';
import SvgIcon from '../SvgIcon.vue';

expect.extend(toHaveNoViolations);

describe('SvgIcon', () => {
  it('should match snapshot', () => {
    const result = shallow(SvgIcon).element;
    expect(result).toMatchSnapshot();
  });

  it('Component throws no accessibility violations', (done) => {
    const html = shallow(SvgIcon).html();
    // pass anything that outputs html to axe
    return axe(html).then((response) => {
      expect(response).toHaveNoViolations();
      done();
    });
  });
});
