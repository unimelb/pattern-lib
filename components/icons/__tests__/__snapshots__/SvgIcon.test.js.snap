// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SvgIcon should match snapshot 1`] = `
<svg
  aria-label=""
  focusable="false"
  height="0"
  role="img"
  width="0"
>
  <use
    xlink:href="#icon-"
  />
   
  <title>
    
  </title>
   
</svg>
`;
